File: Python/Problems/Problems/problem1780.py
import heapq
import math
import sys

import numpy as np

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem1780(Problem):
    def test(self) -> bool:
        n = 91

        expected = True

        result = self.checkPowersOfThree(n)

        return result == expected

    def checkPowersOfThree(self, n: int) -> bool:
        curr = 1
        while curr <= n:
            curr *= 3

        while n > 0:
            if n >= curr:
                n -= curr
            if n >= curr:
                return False
            curr /= 3

        return True

File: C++/Problems/Problems/problem_1780.cpp
#include "problem_1780.h"

bool problem_1780::test() {
    int n = 91;

    bool expected = true;

    auto result = checkPowersOfThree(n);

    return result == expected;
}

bool problem_1780::checkPowersOfThree(int n) {
    int curr = 1;
    while (curr <= n)
        curr *= 3;

    while (n > 0)
    {
        if (n >= curr)
            n -= curr;
        if (n >= curr)
            return false;
        curr /= 3;
    }

    return true;
}
